### PR TITLE
DHFPROD-2962:Add tests to ensure invalid entity model and without baseUri aren't inserted

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
Add tests to ensure invalid entity model and without baseUri aren't inserted
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [ ] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java [hunk 1]
```diff
@@ -42,10 +47,85 @@ public void testDeployTDEWithNoEntities() {
...
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        File entityFile = getHubProject().getHubEntitiesDir().resolve("Customer.entity.json").toFile();
+        try {
+            mapper.writerWithDefaultPrettyPrinter().writeValue(entityFile, mapper.readTree(entity));
+            installUserArtifacts();
+            assertTrue(false, "Inserting entity without baseUri should fail");
```
Comment (akshaysonvane) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java:74]: Can we use `assertThrows` with the Exception that we are expecting? 
Currently, the test will pass in multiple cases like network failure, failure to write the entity file, etc.

Comment (rjrudin) [reply marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java:74]: Use "fail" instead - e.g. fail("Expected a validation error to be thrown").

Note though that a missing baseUri is fine, it defaults to example.org by design.

File: marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java [hunk 2]
```diff
@@ -42,10 +47,85 @@ public void testDeployTDEWithNoEntities() {
...
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        File entityFile = getHubProject().getHubEntitiesDir().resolve("Customer.entity.json").toFile();
+        try {
+            mapper.writerWithDefaultPrettyPrinter().writeValue(entityFile,  mapper.readTree(entity));
+            installUserArtifacts();
+            assertTrue(false, "Inserting invalid entity should fail");
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java:116]: Same thing, fail here.

File: marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java [hunk 3]
```diff
@@ -42,10 +47,85 @@ public void testDeployTDEWithNoEntities() {
...
+        } catch (IOException e) {
+            logger.info("Cannot insert invalid entity");
+        }
+        finally {
+            FileUtils.deleteQuietly(entityFile);
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java:121]: No need to do this, the hub project directory gets cleared out at the start of every test.

File: marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java [hunk 4]
```diff
@@ -42,10 +47,85 @@ public void testDeployTDEWithNoEntities() {
...
+            "    }\n" +
+            "  }\n" +
+            "}";
+        File entityFile = getHubProject().getHubEntitiesDir().resolve("Customer.entity.json").toFile();
+        try {
+            mapper.writerWithDefaultPrettyPrinter().writeValue(entityFile,  mapper.readTree(entity));
+            installUserArtifacts();
+            assertTrue(false, "Inserting invalid entity should fail");
+        } catch (IOException e) {
+            logger.info("Cannot insert invalid entity");
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/entity/EntityServicesAlignmentTest.java:118]: For catch expected exceptions, I recommend: "Caught expected exception due to invalid entity: " + e.getMessage(). 

In fact, it's preferable to make an assertion on the contents of the message so you got the expected error. I'm surprised an IOException is being thrown - don't we want to catch something like an ML FailedRequestException?

**Review by rjrudin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**